packages/two_dimensional_scrollables/example/lib/tree_view/simple_tree.dart

```diff
@@ -0,0 +1,145 @@
+
+/// The class containing a TreeView that highlights the selected row. The
+/// default TreeView.treeNodeBuilder, makes tapping the leading icon of a parent
+/// toggle the node open and closed. The scrollbars will appear as the content
+/// exceeds the bounds of the viewport.
+class TreeExample extends StatefulWidget {
+  /// Creates a screen that demonstrates the TreeView widget.
+  const TreeExample({super.key});
+
+  @override
+  State<TreeExample> createState() => TreeExampleState();
+}
+
+/// The state of the [TreeExample].
+class TreeExampleState extends State<TreeExample> {
+  /// The [TreeViewController] associated with this [TreeView].
+  @visibleForTesting
+  final TreeViewController treeController = TreeViewController();
+
+  /// The [ScrollController] associated with the horizontal axis.
+  @visibleForTesting
+  final ScrollController horizontalController = ScrollController();
+  TreeViewNode<String>? _selectedNode;
+  final ScrollController _verticalController = ScrollController();
+  final List<TreeViewNode<String>> _tree = <TreeViewNode<String>>[
+    TreeViewNode<String>(
+      "It's supercalifragilisticexpialidocious",
+      children: <TreeViewNode<String>>[
+        TreeViewNode<String>(
+          'Even though the sound of it is something quite atrocious',
+        ),
+        TreeViewNode<String>(
+          "If you say it loud enough you'll always sound precocious",
+        ),
+      ],
+    ),
+    TreeViewNode<String>(
+      'Supercalifragilisticexpialidocious',
+      children: <TreeViewNode<String>>[
+        TreeViewNode<String>(
+          'Um-dittle-ittl-um-dittle-I',
+          children: <TreeViewNode<String>>[
+            TreeViewNode<String>(
+              'Um-dittle-ittl-um-dittle-I',
+              children: <TreeViewNode<String>>[
+                TreeViewNode<String>(
+                  'Um-dittle-ittl-um-dittle-I',
+                  children: <TreeViewNode<String>>[
+                    TreeViewNode<String>(
+                      'Um-dittle-ittl-um-dittle-I',
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ],
+        ),
+      ],
+    ),
+  ];
+
+  Map<Type, GestureRecognizerFactory> _getTapRecognizer(
+    TreeViewNode<String> node,
+  ) {
+    return <Type, GestureRecognizerFactory>{
+      TapGestureRecognizer:
+          GestureRecognizerFactoryWithHandlers<TapGestureRecognizer>(
+        () => TapGestureRecognizer(),
+        (TapGestureRecognizer t) => t.onTap = () {
+          setState(() {
+            _selectedNode = node;
+          });
+        },
+      ),
+    };
+  }
+
+  Widget _getTree() {
+    return DecoratedBox(
+      decoration: BoxDecoration(border: Border.all()),
+      child: Scrollbar(
+        controller: horizontalController,
+        thumbVisibility: true,
+        child: Scrollbar(
+          controller: _verticalController,
+          thumbVisibility: true,
```

Comment:
> Do we require the manual addition of Scrollbars for our other scrollable widgets as well?

We do for 2D, pending https://github.com/flutter/flutter/issues/122348 which I don't know when will be reprioritized. On some platforms, Scrollable will call ScrollBehavior.buildScrollbar, otherwise yes they need to be added manually.

In TwoDimensionalScrollable, it does not call ScrollBehavior.buildScrollbar, so there is no case where scrollbars are automatically added in 2D scrolling.